### PR TITLE
New Featured Notebook: Explaining Convolutions

### PR DESCRIPTION
**Pull request by ariguiba**
A request to add a new notebook to the featured notebook.

This notebook is about Convolution of functions in the continuous and the discrete case.

It first introduces a practicle example of noise smoothing (like in signal processing). The user can set different functions and different filter functions and see how the convoluted result changes.

Then, in the second part, convolution of function is explained over a discrete space where the functions are defined on a set of points and with animations and sliders the user can follow back how to the value of the final convoluted function came to be derived from the initial input functions.

### CODE REVIEW

**Comment by ariguiba** [comment]
Hi Fons! This notebook is done, for my part but I am still not a 100% satisfied with the result, as I feel the intuition is still lacking. 

Maybe it's a good idea to put it to review so I can get more feedback and outside ideas about how understandable (or not) it is :+1:

File: src/convolution/convolution_1D.jl [hunk 1]
```diff
@@ -0,0 +1,1757 @@
...
+	md"""Index: $(e_draw)"""
+end
+
+# ╔═╡ 00358bc2-1e49-45a9-9b78-293aadd40976
+md"""## Convolution Explained
+
+It's alright if you can't tell how the resulting function is created. In the following part we will go through each step of the convolution process, to understand better how they work."""
+
+# ╔═╡ f59990f8-ecba-4d16-8b68-f3d40b044d74
+md"""Use the slider below to change the amplitude of the filter function and the range on which it's defined."""
```
Comment (behinger) [inline src/convolution/convolution_1D.jl:61]: I feel like this is too complicated and amplitude modification should be after the explanation nt before

File: src/convolution/convolution_1D.jl [hunk 2]
```diff
@@ -0,0 +1,1757 @@
...
+	xlims!(ax2, (0, 100))
+	ylims!(ax2, (0, 70))
+	
+	# Plotting
+	lines!(ax1, z, fz1, color = "red")
+	lines!(ax1, z, fz2_draw, color = "blue")
+	lines!(ax2, z, fz3, color="gray")
+	lines!(ax2, z, fz3_draw, color="purple")
+
+	f
```
Comment (behinger) [inline src/convolution/convolution_1D.jl:139]: The logic here doesnt really work.

We want the top plot to show the 2 functions (lower left plot now), then we want a multiply sign, then we want the plot with currently the "plus" - indicating we did pointwise multiplications, the we want the result. Else it is quite hard

Also maybe the same scales would help to visualize everyhting better?

**Review by behinger — COMMENTED** [review]
this still needs some work to be useful imho - let's discuss

**Comment by behinger** [comment]
convolution
This goes into the direction of what I envisioned with the idea.

I will think a bit if I can replace the " + + + " with a 90° rotate `}+` to symbolize the addition

**Comment by behinger** [comment]
grafik
I'll stop now, but I think this really helps.

unfortunately, I didnt really understand your variable names etc. so there is a lot of hacking going on :S

**Comment by behinger** [comment]
- I removed the total number of pills, I thought it is a bit confusing?
- I made the text follow the actual number of pills etc.
- I moved the lowest plot so that the day marker aligns.
- I think I improved the emojis by putting a white circle below each, now one can actually recognize it
- I made the markers smaller, and moved them left/right, now the patients/pills dont overlap that much
- 
grafik

- I changed some things in the code, but it is a bit a mess right now :S could probably be simplified quite a bit :S
- I added an easteregg, check out what the patients do once they reached the last pill ;)

**Comment by ariguiba** [comment]
I like the changes a lot it looks great! I added some code refactoring (I focused on the main code by adding comments and left the methods in the appendix for interested users to dive deep into). I couldn't simplify it any further though because we just have a lot of elements in the plot.
Also, I love the easter egg!!

**Comment by fonsp** [comment]
Ohh this is fantastic! I didn't find the easter egg yet 😅

**Pull request merged**